nodebuilder/node/flags.go

```diff
@@ -0,0 +1,79 @@
+package node
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+	flag "github.com/spf13/pflag"
+)
+
+const (
+	// DefaultNetwork is the default network of the current build.
+	DefaultNetwork   = Arabica
+	EnvCustomNetwork = "CELESTIA_CUSTOM"
+	nodeNetworkFlag  = "node.network"
+)
+
+// Flags gives the set of flags that define the network parameters
+func Flags() *flag.FlagSet {
+	flags := &flag.FlagSet{}
+
+	flags.String(
+		nodeNetworkFlag,
+		"",
+		"The name of the network to connect to, e.g. "+listProvidedNetworks(),
+	)
+
+	return flags
+}
+
+// ParseFlags tries to parse the network from the flags and environment,
+// and returns either the parsed network or the build's default network
+func ParseFlags(cmd *cobra.Command) (Network, error) {
+	parsedNetwork := cmd.Flag(nodeNetworkFlag).Value.String()
+	// no network set through the flags, so check if there is an override in the env
+	if parsedNetwork == "" {
+		return ParseNetworkFromEnv()
+	}
+	return Network(parsedNetwork), nil
+}
```

Comment:
`ParseFlags` does not override the `DefaultNetwork`, meaning that flag won't affect our protocol IDs and breaks compatibility with Mamaki. We could still override as a short-term fix. The correct long-term fix here is to make protocol IDs not rely on a constant. Instead, each protocol like HeaderEx should have an option or config through which it can receive and prepend a chain ID in the constructor and not during global initialization.,

Reply:
Ok, another reason why network might be helpful to be in the config no?

Reply:
Or I can pass the network to the modules

Reply:
Network to Modules 👍

Reply:
done btw